### DISCUSSION TITLE
How to find out which location from the target file provided a plug-in?

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-pde/eclipse.pde#897] [General]
Frequently I have to search for "why do we have two versions of this plug-in" problem. Most recently a client had two versions of OSGi (one from an EGit addional library feature and its update site and one from the platform feature. This client used slicer.

My current "process" is to comment out parts of the target platform, reload it and see if it is gone. This is time consuming, as the resolving of the target platform is still relatively slow. Is there an easier way to find out where a plug-in is coming from?

**Comment by merks** [comment]
This sounds like a question/discussion not an issue. When you checkmark show content it shows the answer to your question in the tree on the first tab doesn’t it?

**Reply by vogella** [reply]
This is a discussion, is it not? URL for me is https://github.com/eclipse-pde/eclipse.pde/discussions/897

> When you checkmark show content it shows the answer to your question in the tree on the first tab doesn’t it?

It shows the general available content from this location, not the included plug-ins based on your configuration. So while is is helpful to identify the potential source, it does not answer the question where does "xyz" come from. For example 20 location may have the OSGi bundle included but the features I added from these locations in 18 cases do not include this one.

**Reply by merks** [reply]
Sorry, it's misleading on a phone...

I'm not sure what specifically you are referring to in your comment.  Here it's showing the 14 bundles that come from EMF and not the huge number available at the site:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/aae21fd8-7a0d-4bc4-bc52-c49d2506aad8)

**Reply by vogella** [reply]
Thanks, definitely better than my currently approach, but for bigger sites and , this still requires a lot of checking with standard target editor with scrolling.
E.g nebula with 282 and Sep release with 530 entries.

![image](https://github.com/eclipse-pde/eclipse.pde/assets/139910/ef741042-63d0-4738-a39a-27f8fdd8ede7)

The Repository Explorer looks way better for this task, as it have a filter box, especially with the expert mode which shows technical names. Thanks for that.

Still not the a direct way to answer: "Where does plug-in x come from"? but an easier way to answer "Lets see which locations contain the plug-in X" and once we have check enough locations we can also see where x is coming from.

**Comment by HannesWell** [comment]
Besides what has been said, in the `Content` Tab of the Target Editor you can also `Group by Location`:

![grafik](https://github.com/eclipse-pde/eclipse.pde/assets/44067969/9559400d-0b4b-439b-97d4-bca4136dfaea)


Furthermore if you want to know the dependents/dependencies of a Plugin, the Plug-ins Dependencies view also shows them both to you:

![grafik](https://github.com/eclipse-pde/eclipse.pde/assets/44067969/8a438d93-ccef-4f65-9338-b428abf740c6)

You can open it from the Dependencies Tab in the MANIFEST.MF Editor or directly by clicking on the Focus button ("magnifying glass") in the view.
On a Feature level the `Features` view also shows inter-Feature dependencies.

**Comment by vogella** [comment]
> Besides what has been said, in the Content Tab of the Target Editor you can also Group by Location:

Thank you @HannesWell, this is exactly what I needed. Thanks a bunch, also to @merks for the advice and help
